src/o365/teams/commands/report/teams-report-useractivityuserdetail.ts

```diff
@@ -92,6 +138,28 @@ class TeamsReportUserActivityUserDetailCommand extends GraphCommand {
         return `${args.options.date} is not a valid date. The supported date format is YYYY-MM-DD`;
       }
 
+      if (args.options.output) {
+        if (['csv', 'json', 'text'].indexOf(args.options.output) < 0) {
+          return `${args.options.output} is not a valid output type. The supported values are csv,json,text`;
+        }
+      }
+
+      if (args.options.outputFile && !fs.existsSync(path.dirname(args.options.outputFile))) {
+        return 'Specified outputFile path where to save the file does not exist';
+      }
+
+      if (args.options.outputFile && args.options.output && args.options.output.toLowerCase() === 'csv' && !args.options.outputFile.toLowerCase().endsWith('.csv')) {
```

Comment:
I do not think the extension matters as long as the have `--output` option specified.

Reply:
I have fixed this in the latest commit but do we need to validate the below scenario where output is csv and path ends with json

teams report useractivityuserdetail --period D7 --output **csv** --outputFile '/Users/josephvelliah/Desktop/teams-report-useractivityuserdetail.**json**'

Reply:
My opinion is that we do not have to validate the extension nor the name for the specified file. The user can name it whatever, he/she will get content in `json` format when the `--output` option is set to json.

Reply:
Sure, Thank you for the clarification.